### ISSUE TITLE
Window 86x - ArgumentError: Error #3214

### ISSUE THREAD

**Issue by hj0201** [scout-app/scout-app#173]
When I try to start the exe,  it showed: ArgumentError: Error #146

**Comment by nramin** [comment]
@iamabeagle Thank you!

**Comment by sg1010** [comment]
This is what worked for me in Win8.1

function javaDir() {
if(air.Capabilities.os.match(/Windows/)) {
path = air.File.applicationDirectory.resolvePath("C:\Program Files (x86)\Java\jre1.8.0_25\bin\java.exe");
if(!path.exists){

**Comment by stringyland** [comment]
I applied this fix when I was using Scout in Windows 7, but now I'm on a new computer with a fresh install of Scout, Java and Windows 8 and it no longer works for me. Am going to try a hardcoded path instead, but would love to see this bug resolved (or for Java to stop changing where it keeps things, good luck with that!).

**Comment by eSaner** [comment]
@iamabeagle Thanks! This was super helpful. I just had to change the slashes to forward slashes. Then it worked.

function javaDir()
{
path = air.File.applicationDirectory.resolvePath("C:/ProgramData/Oracle/Java/javapath\java.exe");
return path;
}

**Comment by powerfulsite** [comment]
Hi I'm running into this same issue.  However, when I edit the process_interaction.js per @iamabeagle it won't allow me to save the file.  I get the error:  C:\Program Files(86)\Scout\javascripts\app\process_interaction.js was denied.  I tried editing the slashes a la @eSaner but that didn't work.  Does anyone know how to get around this?

**Comment by eSaner** [comment]
I had the same problem of not being able to save the file. I made a copy of the file and was then able to edit it, and swapped out the original for the copy.

**Comment by powerfulsite** [comment]
@eSaner  Thanks so much for your swift reply!  That worked!

**Comment by wisecrab** [comment]
@eSaner that was the fix I needed! Been having this issue for months and couldn't figure out why this solution didn't help, now it works with the forward slashes! Much appreciated.

**Comment by JDBurnZ** [comment]
This issue exists because either:
* You do not have Java JRE 7 or 6 installed.

**Comment by KaeptnParticle** [comment]
I had a similar issue which happened after an AIR update:

function javaDir()
{
path = air.File.applicationDirectory.resolvePath("C:\\ProgramData\\Oracle\Java\\javapath\java.exe");
return path;
}

Thanks to you, I found the issue fast and removed the "\\", all fine now.

**Comment by realter** [comment]
Thanks @iamabeagle

**Comment by rtro92** [comment]
Thanks @iamabeagle . You solution kept me going! I also had to change the back slashes to forward slashes, and make a copy then save it over the existing one due to Permissions. Cheers!

**Comment by neginparsa** [comment]
Thank you so much @iamabeagle  saved my time!

**Comment by fuad7us** [comment]
Thanks eSaner, It solved after change the slashes to forward slashes and many thanks to iamabeagle.

**Comment by projer** [comment]
Thanks for that, really helped!

Personnally, I have double the \ to make it work :
path = air.File.applicationDirectory.resolvePath("C:\\ProgramData\\Oracle\\Java\\javapath\\java.exe");

Otherwise it escapes the first letters.

**Comment by the1andonlypz** [comment]
Hey Guys,

That works perfectly, but as a note for those on Windows 8, you need to either use forward slashes or double back slashes.

So your file-path would look like one of these:

C:/ProgramData/Oracle/Java/javapath/java.exe

OR

C:\\ProgramData\\Oracle\\Java\\javapath\\java.exe

**Comment by MrADas** [comment]
PLS follow this steps

1.check ur java program into the install programm
![untitled](https://cloud.githubusercontent.com/assets/13918551/9424393/cc6f1e86-4908-11e5-8420-bf780d64cc3c.png)
2.go to the path _ C:\Program Files (x86)\Scout\javascripts\app then open (process_interaction.js) in dreamweaver or notepad,find the code

 function javaDir() {
    if(air.Capabilities.os.match(/Windows/)) {
      path = air.File.applicationDirectory.resolvePath("C:\\Program Files\\Java\\jre7\\bin\\java.exe");
      if(!path.exists){
        path = air.File.applicationDirectory.resolvePath("C:\\Program Files (x86)\\Java\\jre7\\bin\\java.exe");
        if(!path.exists){
          path = air.File.applicationDirectory.resolvePath("C:\\Program Files\\Java\\jre6\\bin\\java.exe");
        }
      }
      return path;
    } else {
      return air.File.applicationDirectory.resolvePath("/usr/bin/java");
    }
  }

Replace this giveen code

function javaDir() {
    if(air.Capabilities.os.match(/Windows/)) {
      path = air.File.applicationDirectory.resolvePath("C:/Program Files/Java/jre7/bin/java.exe");
      if(!path.exists){
        path = air.File.applicationDirectory.resolvePath("C:/Program Files (x86)/Java/jre7/bin/java.exe");
        if(!path.exists){
          path = air.File.applicationDirectory.resolvePath("C:/Program Files (x86)/Java/jre1.8.0_60/bin/java.exe");
        }
      }
      return path;
    } else {
      return air.File.applicationDirectory.resolvePath("/usr/bin/java");
    }
  }

thanks....

**Comment by Eben-Hafkamp** [comment]
woah flawless solution! @iamabeagle

**Comment by dinael** [comment]
Thank @projer  It's work now

**Comment by Shropford** [comment]
Yup... all sorted. (windows 10).. thanks @MrADas  but I made the mistake when I copied and pasted your code to wipe out the remaining closing brackets and semicolon at the very bottom " });"

easily done as you are copy and pasting.... maybe you should add them to your code or add a note of caution.... or just buy me some new glasses :)

**Comment by FTMMM** [comment]
Failed to save 'process_interaction.js': EPERM: operation not permitted, open 'c:\Program Files (x86)\Scout\javascripts\app\proc...

Why I can't save my work ??  I tried to change it but it won't work..

@MrADas

**Comment by the1andonlypz** [comment]
@FTMMM you need to "save as" to a different location (desktop, for example) and then drag the new file into that folder.

**Comment by xaarrssx** [comment]
@FTMMM you need to open the editor as administrator. If you don't grant administrator permission, the OS will no let you write over the file.

**Comment by aaronWcurtis** [comment]
@iamabeagle  Thank you! Like others mentioned, this works for me only after I changed the \ to /
so my JS file looks like the following:
  function javaDir() {
    path = air.File.applicationDirectory.resolvePath("C:/ProgramData/Oracle/Java/javapath/java.exe");
    return path;
  }

**Comment by MrADas** [comment]
thnx..

if any software crack,hardware or network prob pls send me the mail to "
arindam.das.1988.21@gmail.com" or my fb id "
https://www.facebook.com/DrRiversider"...

thank u...

On 15 December 2015 at 08:11, Aaron Curtis <notifications@github.com> wrote:

> @iamabeagle <https://github.com/iamabeagle> Thank you! Like others
> mentioned, this works for me only after I changed the \ to /
> so my JS file looks like the following:
> function javaDir() {
> path =
> air.File.applicationDirectory.resolvePath("C:/ProgramData/Oracle/Java/javapath/java.exe");
> return path;
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mhs/scout-app/issues/173#issuecomment-164625909>.
>



-- 
Mr. Arindam Das
Ph No- 9874142699,9230473651

**Comment by Nezunoban** [comment]
Thanks @iamabeagle!

> function javaDir() {
path = air.File.applicationDirectory.resolvePath("C:/ProgramData/Oracle/Java/javapath/java.exe");
return path;
}

**Comment by TheJaredWilcurt** [comment]
# New Downloads

There is a re-release of the 2012 version of Scout-App (0.7.1) and a new 2016 version of Scout-App 2.

* **[Downloads](https://github.com/TheJaredWilcurt/scout-app/releases)**

**Comment by TheJaredWilcurt** [comment]
@aaronWcurtis @ArtbyArjan @dinael @Eben-Hafkamp @eSaner @FTMMM @fuad7us @hagatorn @hj0201 @iamabeagle @j-rod85dr @JDBurnZ @MrADas @neginparsa @Nezunoban @nramin @pgogy @powerfulsite @projer @realter @rtro92 @Shropford @stringyland @the1andonlypz @wisecrab @xaarrssx 

Wow, that's a lot of people. And I've got good news for all of you! :smile: 

Scout-App is no longer reliant on Adobe Air or Java! Try out the latest version and let us know what you think.

* [Scout-App.io](http://Scout-App.io)

If you have any feedback, create a new issue. Closing this one.

**Comment by joharzmn** [comment]
I have also encountered with this problem while i am making my first project. 
I will solve this issue by just make a copy of my jre folder and rename it by jre7 and it will solve my problem :) Hope this will help you

**Comment by Stelahere** [comment]
i was getting ArgumentError: Error #3214 running on windows 7
tried every suggestions but i was getting same error

After trying changes on  function javaDir()
thanks @iamabeagle 

My solution, I open downloaded file  scout-app-0.7.1-win\Scout-Installer\Air-Java7.exe
run Air-Java7

it worked for me :)
read  scout-app-0.7.1-win\Scout-Installer\_INSTALL INSTRUCTIONS
